.changeset/smooth-cobras-help.md

```diff
@@ -0,0 +1,20 @@
+---
+'astro': minor
+---
+
+Reworks route priority processing to allow for more flexible and intuitive redirects and route injection
```

Comment:
Let me know when this is considered finalized behaviour/naming etc, and please `/ptal` me for a review at that time!